controllers/database.go

```diff
@@ -31,6 +36,53 @@ var dbTemplates = []string{
 	dbSecret,
 }
 
+// extract to var for mocking in testing
+var ConnectAndQueryDatabase = func(host, port, username, password, dbname string) bool {
+	connectionString := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", username, password, host, port, dbname)
+	db, err := sql.Open("mysql", connectionString)
+	if err != nil {
+		return false
+	}
+	defer db.Close()
+
+	testStatement := "SELECT 1;"
+	_, err = db.Exec(testStatement)
+	return err == nil
+}
+
+func (r *DSPAReconciler) VerifyMySQLDBConnection(host, port, username, password, dbname string) bool {
+	return ConnectAndQueryDatabase(host, port, username, password, dbname)
```

Comment:
can you not just remove the receiver on `VerifyMySQLDBConnection`? doesn't seem like it needs to be a method on DSPAReconciler, so you can mock it directly, unless I missed something.

Reply:
yep, already was doing this with the respective Object Store side. updated